lib/srv/app/connections_handler.go

```diff
@@ -384,25 +384,9 @@ func (c *ConnectionsHandler) sessionStartTime(ctx context.Context) time.Time {
 // newTCPServer creates a server that proxies TCP applications.
 func (c *ConnectionsHandler) newTCPServer() (*tcpServer, error) {
 	return &tcpServer{
-		newAudit: func(ctx context.Context, sessionID string) (common.Audit, error) {
-			// Audit stream is using server context, not session context,
-			// to make sure that session is uploaded even after it is closed.
-			rec, err := c.newSessionRecorder(c.closeContext, c.sessionStartTime(ctx), sessionID)
```

Comment:
So this never worked, then? (Just to be certain.)

Reply:
Correct. It used to just create an empty recording with the session ID, which could result in several empty session recordings for a single session. I'm not sure if the recordings even got finalized or just got stuck in the "upload" stage, because we didn't create a session tracker which is used to finalize uploads.